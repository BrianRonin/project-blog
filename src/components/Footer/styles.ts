import styled, { css } from 'styled-components'
import { Main as Text } from '../Text/styles'

export const Main = styled.div`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    background-color: ${theme.colors.white};
    a {
      color: ${theme.colors.darkText};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`
