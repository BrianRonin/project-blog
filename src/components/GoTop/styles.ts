import styled, { css, DefaultTheme } from 'styled-components'

const Light = (theme: DefaultTheme) => css`
  color: ${theme.colors.primary};
  background: ${theme.colors.white};
`
type background = {
  background: boolean
}

export const Main = styled.a<background>`
  ${({ theme, background }) => css`
    position: fixed;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.8;
    border-radius: 50%;
    ${background && Light(theme)};
  `}
`
