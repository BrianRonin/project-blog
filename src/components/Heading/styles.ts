import styled, { css, DefaultTheme } from 'styled-components'
import { HeadingProps } from '.'

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xlarge};
    @media ${theme.media.medium} {
      font-size: ${theme.sizes.large};
    }
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xhuge};
    ${mediaFont(theme)}
  `,
}

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.sizes.xlarge};
  }
`

export const Title = styled.h1<
  Pick<HeadingProps, 'light' | 'size' | 'uppercase'>
>`
  ${({ theme, light, size, uppercase }) => css`
    color: ${light ? theme.colors.white : theme.colors.primary};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${!!size && titleSize[size](theme)};
  `}
`
