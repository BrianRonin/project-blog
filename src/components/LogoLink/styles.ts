import styled from 'styled-components'
export const Main = styled.a`
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;

  > img {
    width: 7rem;

    border-radius: 50%;
  }
`

export const Logo = styled.img`
  max-height: 6rem;
`
