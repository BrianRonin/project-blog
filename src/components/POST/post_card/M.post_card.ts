import { mock_article_header } from '../../ARTICLE/article_header/M.article_header'
export const mock_post_card = {
  slug: 'test',
  ...mock_article_header,
}
