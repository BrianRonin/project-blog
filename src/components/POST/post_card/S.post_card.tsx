import styled, {
  css,
  DefaultTheme,
} from 'styled-components'
import { Title as Heading } from '../../Heading/styles'
import { postCardProps } from './post_card'

const name = (theme: DefaultTheme) => css`
  //
`

export const Main = styled.div`
  ${({ theme }) => css`
    ${Heading} {
      margin: 0;
      margin-top: calc(
        ${theme.spacings.small} - 6px
      );
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    &:hover a {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      opacity: 0.8;
    }
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    transition: opacity 300ms ease-in-out;
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css``}
`
