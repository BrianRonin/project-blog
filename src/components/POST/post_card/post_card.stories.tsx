import {
  Meta,
  Story,
} from '@storybook/react/types-6-0'
import { mock_post_card } from './M.post_card'
import {
  PostCard,
  postCardProps,
} from './post_card'

export default {
  title: 'components/post/Post card',
  component: PostCard,
  args: mock_post_card,
  argTypes: {
    //
  },
} as Meta

export const Template: Story<postCardProps> = (
  args,
) => <PostCard {...args} />

Template.parameters = {
  backgrounds: {
    default: 'light',
  },
}
