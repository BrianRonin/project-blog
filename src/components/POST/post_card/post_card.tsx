import Link from 'next/link'
import { articleHeaderProps } from '../../ARTICLE/article_header/article_header'
import { Heading } from '../../Heading'
import { postProps } from '../post/post'
import * as S from './S.post_card'

export type postCardProps = {
  slug: string
} & articleHeaderProps

export const PostCard = ({
  coverWithMetaProps,
  excerpt,
  title,
  slug,
}: postCardProps) => {
  return (
    <S.Main>
      <Link href={`/post/${slug}`}>
        <a>
          <S.Cover
            src={
              coverWithMetaProps.coverWithMeta.url
            }
            alt={
              coverWithMetaProps.coverWithMeta
                .altText
            }
          ></S.Cover>
        </a>
      </Link>
      <Heading as='h2' size='small'>
        <Link href={`/post/${slug}`}>
          <a>{title}</a>
        </Link>
      </Heading>
      <S.Excerpt>{excerpt}</S.Excerpt>
    </S.Main>
  )
}
