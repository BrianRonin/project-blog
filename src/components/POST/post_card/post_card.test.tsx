import { screen } from '@testing-library/react'
import { renderTheme } from '../../../styles/render-theme'
import { mock_post_card } from './M.post_card'
import { PostCard } from './post_card'
describe('<PostCard />', () => {
  it('should render correctly', () => {
    const { container } = renderTheme(
      <PostCard {...mock_post_card} />,
    )
    expect(
      screen.getByRole('heading', {
        name: mock_post_card.title,
      }),
    )
    expect(
      screen.getByRole('img', {
        name: mock_post_card.coverWithMetaProps
          .coverWithMeta.altText,
      }),
    )
    expect(
      screen.getByText(mock_post_card.excerpt),
    )
    expect(
      screen.getByRole('link', {
        name: mock_post_card.title,
      }),
    ).toHaveAttribute('href', '/post/test')

    expect(container).toMatchSnapshot()
  })
})
