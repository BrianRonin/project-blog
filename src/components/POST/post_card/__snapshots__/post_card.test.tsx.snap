// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should render correctly 1`] = `
.c3 {
  color: #000000;
  text-transform: none;
  font-size: 2.4rem;
}

.c0 .c2 {
  margin: 0;
  margin-top: calc( 1.6rem - 6px );
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover a {
  color: #dc143c;
}

.c0:hover img {
  opacity: 0.8;
}

.c1 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/post/test"
    >
      <img
        alt="um alt image"
        class="c1"
        src="https://miro.medium.com/max/640/0*i1v1In2Tn4Stnwnl.jpg"
      />
    </a>
    <h2
      class="c2 c3"
    >
      <a
        href="/post/test"
      >
        titulo qualquer
      </a>
    </h2>
    <p
      class=""
    >
      uma descrição que vou criando enquanto penso na descrição
    </p>
  </div>
</div>
`;
